PowerJira/public/api-v2/issue/Invoke-JiraGetDeployments.ps1

```diff
@@ -0,0 +1,57 @@
+#https://developer.atlassian.com/cloud/jira/platform/rest/v2/#api-rest-api-2-issue-issueIdOrKey-get
+function Invoke-JiraGetDeployments {
+    param (
+        # The ID of the issue
+        [Parameter(Mandatory,ValueFromPipeline,ValueFromPipelineByPropertyName)]
+        [int32]
+        $Id
+    )
+    begin {
+        $results = @()
+    }
+    process {
+        $functionPath = "/jsw/graphql?operation=DevDetailsDialog"
+        $verb = "POST"
+
+        $query = New-Object RestMethodQueryParams
+
+        $gqlQuery = @'
+query DevDetailsDialog ($issueId: ID!) {
+	developmentInformation(issueId: $issueId){
+		details {
+			deploymentProviders {
+				deployments {
+					displayName
+					url
+					state
+					lastUpdated
+					pipelineId
+					pipelineDisplayName
+					pipelineUrl
+					environment {
+						id
+						type
+						displayName
+					}
+				}
+			}
+		}
+	}
+}
+'@
+
+        $body = New-Object RestMethodJsonBody @{
+            operationName = "DevDetailsDialog"
+            query         = $gqlQuery
+            variables     = @{
+                issueId = $Id
+            }
+        }
+
+        $method = New-Object BodyRestMethod @($functionPath, $verb, $query, $body) 
+        $results += $method.Invoke($JiraContext)
+    }
+    end {
+        $results.data.developmentInformation.details.deploymentProviders.deployments
```

Comment:
This is kinda weird. I followed the 
```
$results = @()
...
$results += $method.Invoke($JiraContext)
...
results.data.developmentInformation.details.deploymentProviders.deployments
```
convention from elsewhere but I think I should be doing this instead(?):
```
$resultsData += $method.Invoke($JiraContext)
$results = $resultsData.data.developmentInformation.details.deploymentProviders.deployments
...
$results
```

Reply:
I think it just depends on what is actually returned from that specific API call, and whether you think anyone will actually want all the wrapper garbage that comes back on a lot of them.  I handled it call by call, so while most of them just return the results directly, there are some that drill into the data object and return only relevant parts, which it looks like is what you might want to do here.